zaza/charm_lifecycle/utils.py

```diff
@@ -528,6 +528,35 @@ def ignore_hard_deploy_errors(
     return False
 
 
+def get_overlay_ppa(yaml_file=None, fatal=True):
+    """Get overlay_ppa from tests_options.
+
+    In the config file for the tests, the tests_options.overlay_ppa option
+    may be used to specify a PPA that will be enabled for all units in
+    the model.
+
+    The tests_options section needs to look like:
+
+        tests_options:
+          overlay_ppa: ppa:ubuntu-security-proposed/ppa
```

Comment:
my only observation is that this should be list of str.

```
overlay_ppa:
  - ppa:foo/ppa
  - ppa:bar/ppa
```

Reply:
I didn't think there would be a use case to use more than one PPA but it might not hurt to code it that way from the start.